extensions/common/aws/aws-s3-core/src/main/java/org/eclipse/edc/aws/s3/AwsClientProviderImpl.java

```diff
@@ -154,11 +151,22 @@ private IamAsyncClient createIamAsyncClient() {
         return builder.build();
     }
 
-    private void handleBaseEndpointOverride(S3BaseClientBuilder<?, ?> builder) {
-        var endpointOverride = configuration.getEndpointOverride();
-        if (endpointOverride != null) {
+    private void handleBaseEndpointOverride(S3BaseClientBuilder<?, ?> builder, String endpointOverride) {
+        URI endpointOverrideUri;
+
+        if (!StringUtils.isBlank(endpointOverride)) {
+            try {
+                endpointOverrideUri = new URI(endpointOverride);
```

Comment:
Using `URI.create(endpointOverride)` there will be no need to catch the exception

Reply:
Done.